prompting/validators/tasks.py

```diff
@@ -0,0 +1,181 @@
+            f"- {criterion.compose_text()}" for criterion in self.criteria
+        ]
+        criteria_bullet_points_str = "\n".join(criteria_bullet_points)
+
+        prompt_template = textwrap.dedent(
+            """\
+        Your task is to summarize the text delimited with triple backticks:
+        '''{base_text}'''
+        
+        The following criteria must be respected:
+        {criteria}
+        - Do not try to create questions or answers for your summarization. 
+        """
+        )
+
+        prompt = prompt_template.format(
+            base_text=self.base_text, criteria=criteria_bullet_points_str
+        )
+        return prompt
+
+
+class QuestionGenerationTask(Task):
+    def compose_prompt(self) -> str:
+        # Aggregates criteria in bullet points
+        criteria_bullet_points = [
+            f"- {criterion.compose_text()}" for criterion in self.criteria
+        ]
+        criteria_bullet_points_str = "\n".join(criteria_bullet_points)
+
+        prompt_template = textwrap.dedent(
+            """\
+        Your task is to ask a single relevant and insightful question about the preceding context delimited with triple backticks:
+        '''{base_text}'''
+        
+        The following criteria must be respected:
+        {criteria}                                      
+        - Do not answer the question you generate. 
+        - Do not try to summarize the text
+        """
+        )
+
+        prompt = prompt_template.format(
+            base_text=self.base_text, criteria=criteria_bullet_points_str
+        )
+        return prompt
+
+
+class QuestionAnswerTask(Task):
+    def compose_prompt(self) -> str:
+        # Aggregates criteria in bullet points
+        criteria_bullet_points = [
+            f"- {criterion.compose_text()}" for criterion in self.criteria
+        ]
+        criteria_bullet_points_str = "\n".join(criteria_bullet_points)
+
+        prompt_template = textwrap.dedent(
+            """\
+        Read the preceding context delimited with triple backticks carefully.
+        Your task is to provide a step-by-step answer to the last question found in the text, elaborating on your thought process:
+        '''{base_text}'''
+        
+        The following criteria must be respected:
+        {criteria}                                      
+        - Do not include questions or summaries in your answer.
+        """
+        )
+
+        prompt = prompt_template.format(
+            base_text=self.base_text, criteria=criteria_bullet_points_str
+        )
+        return prompt
+
+
+def create_summarization_task(base_text: str) -> SummaryTask:
+    match_words_criteria = MatchLengthCriteria(
+        penalty=0.1,
+        target_length=random.randint(50, 200),
+        unit=TextLengthUnitEnum.WORDS,
+    )
+    match_length_criteria = MatchLengthCriteria(
+        penalty=0.1,
+        target_length=random.randint(4, 8),
+        unit=TextLengthUnitEnum.SENTENCES,
```

Comment:
An average sentence is 15-20 words. This means that length criteria are actually somewhat inconsistent. This means you may introduce a large probability of incurring a penalty for one MatchLengthCriteria when correctly following the other MatchLengthCriteria.

I'm also unsure if it makes sense to stack multiple length criteria as it will probably just multiply the penalty (assuming they are aligned correclty).
Instead just sample one of them.